### ISSUE TITLE
Rails 7.1 + query_constraints: breaks preloader

### ISSUE THREAD

**Issue by yosiat** [crashtech/torque-postgresql#90]
Hi,

I am working on introducing `query_constraints` usage in my project, but preloading isn't working when `torque-postgresql` is used.

Error:
```
/Users/yosi/.rvm/gems/ruby-3.2.2/gems/torque-postgresql-3.3.0/lib/torque/postgresql/associations/preloader/loader_query.rb:27:in `connected_through_array?': undefined method `array?' for nil:NilClass (NoMethodError)

            foreign_column.array?
                          ^^^^^^^
        from /Users/yosi/.rvm/gems/ruby-3.2.2/gems/torque-postgresql-3.3.0/lib/torque/postgresql/associations/preloader/loader_query.rb:18:in `query_condition_for'
        from /Users/yosi/.rvm/gems/ruby-3.2.2/gems/torque-postgresql-3.3.0/lib/torque/postgresql/associations/preloader/loader_query.rb:13:in `load_records_for_keys'
        from /Users/yosi/.rvm/gems/ruby-3.2.2/gems/activerecord-7.1.2/lib/active_record/associations/preloader/association.rb:89:in `load_records'
        from /Users/yosi/.rvm/gems/ruby-3.2.2/gems/activerecord-7.1.2/lib/active_record/associations/preloader/association.rb:68:in `records'
        from /Users/yosi/.rvm/gems/ruby-3.2.2/gems/activerecord-7.1.2/lib/active_record/associations/preloader/association.rb:28:in `records_for'
        from /Users/yosi/.rvm/gems/ruby-3.2.2/gems/activerecord-7.1.2/lib/active_record/associations/preloader/association.rb:32:in `load_records_in_batch'
```

Code:
```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "activerecord", "~> 7.1.0"
  gem "pg"
  gem "torque-postgresql", git: "https://github.com/crashtech/torque-postgresql", branch: "master"
end

require "active_record"
require "logger"
require "torque-postgresql"

ActiveRecord::Base.establish_connection(
  adapter: "postgresql",
  database: "test",
  encoding: "unicode",
  host: "localhost",
  port: "5432",
  password: "12345",
  username: "test")

ActiveRecord::Schema.define do
  create_table :comments, force: true do |t|
    t.integer :tenant_id
    t.integer :author_id
  end

  create_table :authors, force: true do |t|
    t.integer :tenant_id
  end
end

class Author < ActiveRecord::Base
  query_constraints :tenant_id, :id

  has_many :comments
end

class Comment < ActiveRecord::Base
  query_constraints :tenant_id, :id

  belongs_to :author
end

author = Author.create!
author.comments << [Comment.new, Comment.new]

ActiveRecord::Associations::Preloader.new(records: Author.all, associations: [:comments]).call
```

**Comment by crashtech** [comment]
This is now fixed on version 3.3.1.

**Comment by yosiat** [comment]
Thanks for the fix and release 🙏
